clippy_lints/src/misc.rs

```diff
@@ -623,15 +623,14 @@ fn check_cast(cx: &LateContext<'_, '_>, span: Span, e: &Expr, ty: &Ty) {
     if_chain! {
         if let TyKind::Ptr(MutTy { mutbl, .. }) = ty.kind;
         if let ExprKind::Lit(ref lit) = e.kind;
-        if let LitKind::Int(value, ..) = lit.node;
-        if value == 0;
+        if let LitKind::Int(0, _) = lit.node;
         if !in_constant(cx, e.hir_id);
         then {
-            let msg = match mutbl {
-                Mutability::MutMutable => "`0 as *mut _` detected. Consider using `ptr::null_mut()`",
-                Mutability::MutImmutable => "`0 as *const _` detected. Consider using `ptr::null()`",
+            let (msg, sugg) = match mutbl {
+                Mutability::MutMutable => ("`0 as *mut _` detected. Consider using `ptr::null_mut()`", "std::ptr::null_mut()"),
+                Mutability::MutImmutable => ("`0 as *const _` detected. Consider using `ptr::null()`", "std::ptr::null()"),
             };
-            span_lint(cx, ZERO_PTR, span, msg);
+            span_lint_and_sugg(cx, ZERO_PTR, span, msg, "try", sugg.to_string(), Applicability::MaybeIncorrect);
```

Comment:
What's the reason for marking this as `MaybeIncorrect`?

Reply:
Because if we replace `let _ = 0 as *mut f64;` with `let _ = std::ptr::null();` it would be compile error.

Reply:
Ok, I think I would try to get span of `*mut f64` to suggest the right thing.

Reply:
Oh yeah. We can also merge this as is, building a `MachineApplicable` suggestion seems to be a tedious task.

Can you add a note, that this is `MaybeIncorrect`, because type inference may not work with the suggested code?

Reply:
I have done the tedious task! Could you re review?

Reply:
Oops: O_o. I was wrong! The suggestion is wrong.